### DISCUSSION TITLE
Marker `onPress` event is not matching event type def

### DISCUSSION THREAD

**Discussion by lucksp** [react-native-maps/react-native-maps#5018] [Q&A]
I am trying to render a custom tooltip positioned properly above the marker as `Callout` doesn't seem to handle centering well.  I have custom markers and am hoping to store the `MarkerPressEvent#position` to state which will position the tooltip.

```js
return (
            <Marker
              coordinate={{ latitude: location.lat, longitude: location.lng }}
              identifier={datum.id}
              key={datum.id}
              onPress={event => console.log(event)} // this event doesn't match `MarkerPressEvent`
            >
              <View>
                {clickedMarker === i && (
                  <Callout style={[mapStyles.callout]} tooltip>
                    <View>
                      <Typography>{datum.locationName}</Typography>
                    </View>
                    <View style={mapStyles.triangle}></View>
                  </Callout>
                )}
                <MapPinIcon />
              </View>
            </Marker>
```

However, when I try to access the `position` value for that marker, it's undefined.  I also tried the `onMarkerPress` event from the `MapView` itself and this too is undefined.  

Am I doing something wrong in terms of accessing the `MarkerPressEvent`?

**Comment by carolwangg** [comment]
For anyone else trying to find a solution to this issue, just call `onPress={event => console.log(event.nativeEvent)}` and that should give all the props specified in the MarkerPressEvent type. Make sure you set `identifier={id}`, that will be what is returned to you for the `id` field.
